frontend/src/store/LoadingState.ts

```diff
@@ -6,11 +6,17 @@ import { ref } from 'vue';
     Components can read this indicator but *-DON'T-* need to set the value since it is being
     taken care by axios interceptors.
 
-    If http is being triggered by third-party libraries, loading will need to be manually 
+    If http is being triggered by third-party libraries, loading will need to be manually
     set (not part of app's axios instance).
 */
-const isLoading = ref<boolean>(false);
+const loadingState = ref<boolean>(false);
 
-export default {
-    isLoading,
-};
+// Refactor adding setter and getter so they can be be used in testing as well
+
+export const setLoadingState = (status: boolean) => {
+    loadingState.value = status;
+}
+
+export const isLoading = () => {
```

Comment:
Hi Nick, I feel normally we call it getLoadingState? so set/get/reset/remove ?

Reply:
I felt that `isLoading` would make more sense understanding what we are doing throughout the code, but you are right, it's not the usual pattern. I'll change it to getLoadingState.

Reply:
This is a state with Boolean value and the exported function already retrieves the `.value` from the state for the need. I think it should be ok for calling `isLoading`, so the usage is more readable on the javascript or html condition.